tutorials/generative/realism_diversity_metrics/realism_diversity_metrics.py

```diff
@@ -0,0 +1,356 @@
+)
+val_ds = Dataset(data=val_datalist, transform=val_transforms)
+val_loader = DataLoader(val_ds, batch_size=180, shuffle=True, num_workers=4)
+
+# +
+# Create some synthetic data for visualisation
+n_synthetic_images = 3
+noise = torch.randn((n_synthetic_images, 1, 64, 64))
+noise = noise.to(device)
+scheduler.set_timesteps(num_inference_steps=50)
+unet.eval()
+
+with torch.no_grad():
+    syn_images, intermediates = inferer.sample(input_noise=noise, diffusion_model=unet,
+                                              scheduler=scheduler,save_intermediates=True,
+                                              intermediate_steps=100)
+
+# Plot 3 examples from the synthetic data
+fig, ax = plt.subplots(nrows=1, ncols=3)
+for image_n in range(3):
+    ax[image_n].imshow(syn_images[image_n, 0, :, :].cpu(), cmap="gray")
+    ax[image_n].axis("off")
+# -
+
+# ## Compute FID
+
+# The FID measures the distance between the feature vectors from the real images and those obtained from generated images. In order to compute the FID the images need to be passed into a pre-trained network to get the desired feature vectors. Although the FID is commonly computed using the Inception network, here, we used a pre-trained version of the RadImageNet to calculate the feature space. Lower FID scores indicate that the images are more similar, with a perfect score being 0 indicating that the two groups of images are identical.
+
+radnet = torch.hub.load("Warvito/radimagenet-models", model="radimagenet_resnet50", verbose=True)
+radnet.to(device)
+radnet.eval()
+
+# Here, we will load the real and generate synthetic images from noise and compute the FID of these two groups of images. Because we are generating the synthetic images on this code snippet the entire cell will take about 6 mins run and most of the this time is spent in generating the images. The loading bars show how long it will take to complete the image generation for each mini-batch.
+
+# +
+fid_scores = []
+unet.eval()
+
+for step, x in enumerate(val_loader):
+    # Get the real images
+    real_images = x["image"].to(device)
+
+    # Generate some synthetic images using the defined model
+    n_synthetic_images = len(x['image'])
+    noise = torch.randn((n_synthetic_images, 1, 64, 64))
+    noise = noise.to(device)
+    scheduler.set_timesteps(num_inference_steps=25)
+
+    with torch.no_grad():
+        syn_images, intermediates = inferer.sample(input_noise=noise, diffusion_model=unet,
+                                                  scheduler=scheduler,save_intermediates=True,
+                                                  intermediate_steps=100)
+
+        # Get the features for the real data
+        real_eval_feats = get_features(real_images)
+
+        # Get the features for the synthetic data
+        synth_eval_feats = get_features(syn_images)
+
+        fid = FIDMetric()
+        fid_res = fid(synth_eval_feats.to(device), real_eval_feats.to(device))
+        fid_scores.append(fid_res)
```

Comment:
Usually, FIDs are not computed as a mean of batches. It is evaluating the similarity between 2 distributions. consider to accumulate the features of the two groups across all batches and them compute a single FID score